### ISSUE TITLE
Question: Compatibility with .NET Standard projects

### ISSUE THREAD

**Issue by NoTuxNoBux** [MariusVolkhart/Nopen.NET#4]
Hi

Thanks for this neat package!

I see that there is an analyzer DLL and a runtime DLL. I'm assuming that the latter is what contains the `OpenAttribute` because code bases that use this analyzer need to be able to add said attribute to classes so the Roslyn analyzer can pick it up later to mark classes as explicitly open. Is it possible, however, that the DLL for the `OpenAttribute` _also_ erroneously contains the analyzer code? To visually state what I mean:

* Roslyn Analyzer DLL (`analyzers` subfolder of the NuGet package)
    * `OpenAttribute.cs`
    * `OpenClassAnalyzer.cs`
    * `OpenClassCodeFixProvider.cs`
    * ...
* Code DLL (`lib` subfolder of the NuGet package)
    * `OpenAttribute.cs`
    * `OpenClassAnalyzer.cs` (should this be present?)
    * `OpenClassCodeFixProvider.cs` (should this be present?)

I notice this because I'm trying to use both analyzer and library in a Unity project, which uses .NET Standard 2.1. The analyzer runs fine, but Unity is complaining about the accompanying library:

```
Assembly 'Assets/Packages/NuGet/nopen.net/1.1.0-beta.23/lib/netstandard2.0/Nopen.NET.dll' will not be loaded due to errors:
Unable to resolve reference 'Microsoft.CodeAnalysis'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Nopen.NET references strong named System.Collections.Immutable Assembly references: 1.2.3.0 Found in project: 6.0.0.0.
Assembly Version Validation can be disabled in Player Settings "Assembly Version Validation"
Unable to resolve reference 'Microsoft.CodeAnalysis.CSharp'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'Microsoft.CodeAnalysis.Workspaces'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'System.Composition.AttributedModel'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
```

I suspect the above problem since the `OpenAttribute` is not using said classes - only the analyzer is. IIRC immutable collections also appear to only be part of .NET Core, not .NET Standard, and [require a shim](https://www.nuget.org/packages/System.Collections.Immutable/7.0.0-preview.7.22375.6#readme-body-tab) on said projects, explaining the above errors.

**Comment by NoTuxNoBux** [comment]
In a similar fashion, I can work around the problem by dropping the library DLL and dropping `OpenAttribute` in my Unity code-base directly, in which case the analyzer works as expected.

**Comment by MariusVolkhart** [comment]
Been a while since I’ve worked in .NET, but yes, I concur with your analysis and ideally the analyzers wouldn’t be part of the runtime dependencies. I’d have to look into how to configure the packages to make that work. I’m also open to raising the minimum SDK version of that’s necessary. Are you interested in fixing this yourself?

**Comment by NoTuxNoBux** [comment]
Thanks for the response.

> Are you interested in fixing this yourself?

I am, but can't make a promise as to when or how fast I will be able to get to it (just to manage expectations). Feel free to leave it open if you're not interested in fixing this yourself (I understand); just knowing that my understanding was correct already helps, this means I know what path to take forward when I (or someone else) gets around to fixing it :slightly_smiling_face: .

**Comment by NoTuxNoBux** [comment]
I found some time and proposed a fix in #5.
